tests/rptest/scale_tests/tiered_storage_single_partition_test.py

```diff
@@ -203,6 +226,13 @@ def metadata_readable():
         top_segment = list(manifest['segments'].values())[-1]
         offset_delta = top_segment['delta_offset_end']
         segment_count = len(manifest['segments'])
+
+        spill_manifests = bucket.get_spillover_manifests(
+            NTP(ns='kafka', topic=self.topic, partition=0))
+        if spill_manifests is not None:
+            for spill_meta, spill_manifest in spill_manifests.items():
+                segment_count += len(spill_manifest['segments'])
```

Comment:
nit: the segment count for archive/stm region could be extracted in a BucketView util